src/openeo_gfmap/utils/catalogue.py

```diff
@@ -211,26 +215,46 @@ def s1_area_per_orbitstate_vvvh(
     spatial_extent = box(*bounds)
 
     # Computes if there is the full overlap for each of those states
-    union_ascending = unary_union(ascending_products)
-    union_descending = unary_union(descending_products)
+    union_ascending = unary_union(ascending_products[0])
+    union_descending = unary_union(descending_products[0])
 
     ascending_covers = union_ascending.contains(spatial_extent)
     descending_covers = union_descending.contains(spatial_extent)
 
+    # Computes max temporal gap. Include requested start and end date so we dont miss
+    # any start or end gap before first/last observation
+    ascending_timestamps = pd.DatetimeIndex(
+        sorted(
+            [pd.to_datetime(temporal_extent.start_date, utc=True)]
+            + ascending_products[1]
+            + [pd.to_datetime(temporal_extent.end_date, utc=True)]
+        )
+    )
+
+    descending_timestamps = pd.DatetimeIndex(
+        sorted(
+            [pd.to_datetime(temporal_extent.start_date, utc=True)]
+            + descending_products[1]
+            + [pd.to_datetime(temporal_extent.end_date, utc=True)]
+        )
+    )
+
     # Computes the area of intersection
     return {
         "ASCENDING": {
             "full_overlap": ascending_covers,
+            "max_temporal_gap": ascending_timestamps.diff().max().days,
```

Comment:
I guess this will break if you only request data for a single day? since you cannot compute the diff. 

is there any safeguard in place to capture this behavior?

Reply:
okay perfect!